keras/models.py

```diff
@@ -89,7 +89,12 @@ def compile(self, optimizer, loss, class_mode="categorical", theano_mode=None):
             raise Exception("Invalid class mode:" + str(class_mode))
         self.class_mode = class_mode
 
-        updates = self.optimizer.get_updates(self.params, self.regularizers, self.constraints, train_loss)
+        if hasattr(self, 'cost_updates'):
+            for u in self.cost_updates:
+                if u.modifies_cost:
+                    train_loss = u(train_loss)
```

Comment:
You can get rid of this parameter (and its check in `compile`) by using:
`train_loss += u.loss() `

(which would return 0 if the update does not affect the loss).